GenderPayGap.WebUI/Controllers/Admin/AdminRemoveUserController.cs

```diff
@@ -0,0 +1,130 @@
+using GenderPayGap.Database;
+using GenderPayGap.Extensions;
+using GenderPayGap.Extensions.AspNetCore;
+using GenderPayGap.WebUI.Classes;
+using GenderPayGap.WebUI.Helpers;
+using GenderPayGap.WebUI.Models.Admin;
+using GenderPayGap.WebUI.Services;
+using GovUkDesignSystem.Parsers;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace GenderPayGap.WebUI.Controllers
+{
+    [Authorize(Roles = "GPGadmin")]
+    [Route("admin")]
+    public class AdminRemoveUserController : Controller
+    {
+
+        private readonly AuditLogger auditLogger;
+        private readonly IDataRepository dataRepository;
+        private readonly IRegistrationRepository RegistrationRepository;
+
+        public AdminRemoveUserController(IDataRepository dataRepository,
+            IRegistrationRepository registrationRepository,
+            AuditLogger auditLogger)
+        {
+            this.dataRepository = dataRepository;
+            RegistrationRepository = registrationRepository;
+            this.auditLogger = auditLogger;
+        }
+
+        [HttpGet("/remove-user-organisation")]
+        public IActionResult RemoveUserFromAnOrganisation(long orgId, long userId, bool fromViewUserPage)
+        {
+            UserOrganisation userOrg = dataRepository.GetAll<UserOrganisation>()
+                .FirstOrDefault(u => u.UserId == userId && u.OrganisationId == orgId);
+
+            ValidateUserOrganisationForRemoval(userOrg);
+
+            var viewModel = new AdminRemoveUserViewModel
+            {
+                OrganisationId = userOrg.OrganisationId,
+                OrganisationName = userOrg.Organisation.OrganisationName,
+                UserId = userOrg.UserId,
+                UserFullName = userOrg.User.Fullname,
+                FromViewUserPage = fromViewUserPage
+            };
+            return View("ConfirmRemoving", viewModel);
+        }
+
+        [PreventDuplicatePost]
+        [ValidateAntiForgeryToken]
+        [HttpPost("/remove-user-organisation")]
+        public async Task<IActionResult> RemoveUserFromAnOrganisation(AdminRemoveUserViewModel viewModel)
+        {
+            var user = dataRepository.Get<User>(viewModel.UserId);
+            var organisation = dataRepository.Get<Organisation>(viewModel.OrganisationId);
+            UserOrganisation userOrg = dataRepository.GetAll<UserOrganisation>()
+                .FirstOrDefault(u => u.UserId == viewModel.UserId && u.OrganisationId == viewModel.OrganisationId);
+
+            ValidateUserOrganisationForRemoval(userOrg);
+
+            viewModel.ParseAndValidateParameters(Request, m => m.Reason);
+
+            if (viewModel.HasAnyErrors())
+            {
+                // If there are any errors, return the user back to the same page to correct the mistakes
+                return View("ConfirmRemoving", viewModel);
+            }
+
+            await RegistrationRepository.RemoveRegistrationAsync(userOrg, user);

```

Comment:
(I wouldn't expect you to know this, but...) I think this `RegistrationRepository.RemoveRegistrationAsync` method does:
* more than we need it to do
* and, simultaneously, not everything that we need it to!

It:
* Checks some things for null (which, for us, won't be the case)
* Removes the UserOrg from a list on the Org (but doesn't remove the UserOrg from a similar list on the User (which makes me wonder whether that's necessary at all))
* Changes the Org.LatestRegistration (but there's a case where it might not change it and we'd get an error!)
* Logs some things to a CSV file (which we happen to be moving away from)
* Deletes the UserOrg

I reckon we could reduce this to:
* Change Org.LatestRegistertion to be null (we don't use it for anything important anyway)
* Delete the UserOrg
* Save changes

Then we could remove the dependency on the RegistrationRepository